### PR TITLE
File bugs against TypeScript for the workarounds in transformer mode.

### LINKED ISSUE #528: File issues against TypeScript for the discovered bugs with transformers
Placeholder issue that is linked from source code.

### PR DESCRIPTION
**Pull request by tbosch**
Closes #528

### CODE REVIEW

File: src/decorator-annotator.ts [hunk 1]
```diff
@@ -147,6 +147,8 @@ export class DecoratorClassVisitor {
    *
    * This method looks for the place where the value that is associated to
    * the type is defined and returns that identifier instead.
+   *
+   * This migth be simplified when https://github.com/Microsoft/TypeScript/issues/17516 is solved.
```
Comment (evmar) [inline src/decorator-annotator.ts]: Typo: "might"

File: src/transformer_util.ts [hunk 2]
```diff
@@ -603,16 +610,11 @@ export function getMutableClone<T extends ts.Node>(node: T): T {
 }
 
 /**
- * This is a version of `ts.visitEachChild` that does not visit children of types
- * to prevent errors from TypeScript.
- * TODO(tbosch): file an issue with TypeScript, tracked in
- * https://github.com/angular/tsickle/issues/528.
+ * This is a version of `ts.visitEachChild` that does not visit children of types,
+ * as this lead to errors in TypeScript < 2.4.0 and as they are not printed anyways.
```
Comment (evmar) [inline src/transformer_util.ts]: typo: leads

also, maybe "visited" instead of "printed" (?)

Comment (tbosch) [inline src/transformer_util.ts]: I meant `emitted`, i.e. as types are not emitted.

**Comment by tbosch** [comment]
@evmar PTAL

**Pull request merged**